crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -1587,55 +1596,103 @@ impl std::fmt::Display for PubGrubHint {
                 package,
                 version,
                 tags,
+                closest,
             } => {
                 let s = if tags.len() == 1 { "" } else { "s" };
-                write!(
-                    f,
-                    "{}{} Wheels are available for `{}` ({}) with the following Python tag{s}: {}",
-                    "hint".bold().cyan(),
-                    ":".bold(),
-                    package.cyan(),
-                    format!("v{version}").cyan(),
-                    tags.iter()
-                        .map(|tag| format!("`{}`", tag.cyan()))
-                        .join(", "),
-                )
+                if let Some((py, abi, platform)) = closest {
+                    write!(
+                        f,
+                        "{}{} Wheels are available for `{}` ({}) with the following Python tag{s}: {}. The closest match is `{}`.",
+                        "hint".bold().cyan(),
+                        ":".bold(),
+                        package.cyan(),
+                        format!("v{version}").cyan(),
+                        tags.iter()
+                            .map(|tag| format!("`{}`", tag.cyan()))
+                            .join(", "),
+                        format!("{py}-{abi}-{platform}").cyan(),
+                    )
+                } else {
+                    write!(
+                        f,
+                        "{}{} Wheels are available for `{}` ({}) with the following Python tag{s}: {}",
+                        "hint".bold().cyan(),
+                        ":".bold(),
+                        package.cyan(),
+                        format!("v{version}").cyan(),
+                        tags.iter()
+                            .map(|tag| format!("`{}`", tag.cyan()))
+                            .join(", "),
+                    )
+                }
             }
             Self::AbiTags {
                 package,
                 version,
                 tags,
+                closest,
             } => {
                 let s = if tags.len() == 1 { "" } else { "s" };
-                write!(
-                    f,
-                    "{}{} Wheels are available for `{}` ({}) with the following ABI tag{s}: {}",
-                    "hint".bold().cyan(),
-                    ":".bold(),
-                    package.cyan(),
-                    format!("v{version}").cyan(),
-                    tags.iter()
-                        .map(|tag| format!("`{}`", tag.cyan()))
-                        .join(", "),
-                )
+                if let Some((py, abi, platform)) = closest {
+                    write!(
+                        f,
+                        "{}{} Wheels are available for `{}` ({}) with the following ABI tag{s}: {}. The closest match is `{}`.",
```

Comment:
I think @konsti had a nice suggestion, to say something like: "A wheel exists for a different architecture" (https://github.com/astral-sh/uv/issues/2777#issuecomment-2588264790)